third/afs-krb5/src/aklog.1
.\" 
.\" $Id: aklog.1,v 1.1.1.1 2003-02-13 00:12:35 zacheiss Exp $
.\"
.\" Copyright 1990,1991 by the Massachusetts Institute of Technology
.\" For distribution and copying rights, see the file "mit-copyright.h"
.\"
.TH AKLOG 1 "August 1, 1990"
.UC 4
.SH NAME
aklog \- Obtain tokens for authentication to AFS

.SH SYNOPSIS
.B aklog 
[ 
.B \-d 
] [ 
.B \-hosts
] [ 
.B \-zsubs 
] [ 
.B \-noprdb 
] [ [
.B \-cell | \-c 
] cell [
.B \-k 
kerberos-realm
] ] [ [ 
.B \-path | \-p 
] pathname ]

.SH DESCRIPTION
The 
.I aklog
program is used to authenticate to a cell or directory in 
.I AFS,
the Andrew Filesystem, by obtaining AFS 
.I tokens.  
Ordinarily, aklog is not used directly but called by
.IR attach (1).

If 
.I aklog
is invoked with no command line arguments, it will obtain tokens 
for the workstation's local cell.  It is possible to invoke
.I aklog
with arbitrarily many cells and pathnames
specified on the command line.  
.I aklog 
knows how to expand cell name abbreviations, so short
forms of cell names can be use used.  In addition, 
.I aklog
understands the following command line options:

.TP 4 
.B  \-cell | \-c \fRcell
This flag is not ordinarily necessary since 
.I aklog
can usually figure out when an argument is a cell.  It can be
used to introduce a cell name that would ordinarily be mistaken
for a path name if this should be required.  If this flag is
omitted, an argument will be treated as a cell name if it
contains no slashes (/) and is neither "." nor ".." .

.TP
.B \-k \fRkerberos-realm
This flag is valid only when immediately following the name of a
cell.  It is used to tell 
.I aklog
what kerberos realm should be used while authenticating to the
preceding cell.  This argument is unnecessary except when the
workstation is not properly configured.  Ordinarily, 
.I aklog
can determine this information on its own.

.TP
.B \-path | \-p \fRpathname
Like the \-cell flag, this flag is usually unnecessary.  When it
appears, the next command line argument is always treated as a
path name.  Ordinarily, an argument is treated as a path name
if it is "." or ".." or if it contains a slash (/).  

.TP
.B \-hosts
Prints all the server addresses which may act as a single point of
failure in accessing the specified directory path.  Each element of the
path is examined, and as new volumes are traversed, if they are not
replicated, the server's IP address containing the volume will be
displayed.
.IR Attach (1)
invokes 
.I aklog
with this option.  The output is of the form

host: 
.I IP address

.TP
.B \-zsubs
Causes the printing of the zephyr subscription information that a
person using a given path or cell would want.  
.IR Attach (1)
invokes 
.I aklog
with this option.  The output is of the form

zsub: 
.I instance

where 
.I instance
is the instance of a class filsrv zephyr subscription.

.TP
.B \-noprdb
Ordinarily, aklog looks up the AFS ID corresponding to the name
of the person invoking the command.  Specifying this flag turns
off this functionality.  This may be desirable if the protection
database is unavailable for some reason and tokens are desired
anyway.  

.TP
.B \-d
Turns on printing of debugging information.  This option is not
intended for general users.

.SH EXIT CODES
The exit status of
.I aklog
will be one of the following:
.TP 5
0
Success -- No error occurred.
.TP 5
1
Usage -- Bad command syntax; accompanied by a usage message.
.TP 5
2
Something failed -- More than one cell or pathname was given on
the command line and at least one failure occurred.
A more specific error status is returned when only one directive
is given.
.TP 5
3
AFS -- Unable to get AFS configuration or unable to get
information about a specific cell.
.TP 5
4
Kerberos -- Unable to get tickets for authentication.
.TP 5
5
Token -- Unable to get tokens.
.TP 5
6
Bad pathname -- The path given was not a directory or 
.I lstat(2)
failed on some component of the pathname.
.TP 5
7
Miscellaneous -- An internal failure occurred.  For example, 
.I aklog
returns this if it runs out of memory.

.SH EXAMPLES
.br
To get tokens for the local cell:
.br
% aklog

To get tokens for the athena.mit.edu cell:
.br
% aklog athena.mit.edu
.br
or
.br
% aklog athena

To get tokens adequate to read /afs/athena.mit.edu/user/p/potato:
.br
% aklog /afs/athena.mit.edu/user/p/potato

To get tokens for a test cell that is in a test Kerberos realm:
.br
% aklog testcell.mit.edu -k TESTREALM.MIT.EDU

.SH SEE ALSO
attach(1), tokens(1), unlog(1)

.SH AUTHOR
Emanuel Jay Berkenbilt (MIT-Project Athena)


doc/news
This file is intended to keep track of user-visible changes to the
release, for the construction of the Athena release notes.

Changes from release 8.0 to release 8.1:

	* Athena version numbers are now dotted triplets, e.g. 8.1.0
	  or 8.1.12.  Paths which include all or part of the Athena
	  version number now take the form "pathname-version",
	  e.g. "srvd-8.1" or "dash.menus-8.1"

	* In release 8.0, running "xmkmf" with the default path would
	  generate a Makefile which looked for its include files and
	  libraries in /mit/x11, which could be one of several
	  lockers.  Now it is preferrable to use the vendor X
	  libraries, so this bit of glue is going away.  There will be
	  no /usr/athena/bin/xmkmf any more, nor any symlinks in
	  /usr/athena pointing to the x11 locker.

	  Users can still do something like "add -f x11r5" and then
	  "xmkmf", and things should work, if the x11r5 locker is
	  built for their platform.

	* Athena bitmaps have moved to /usr/athena/include/X11/bitmaps.
	  Previously, they were in /usr/athena/lib/X11/bitmaps before
	  because of the /mit/x11 glue.

	* The symlink /bin/csh -> /bin/athena/tcsh has gone away.  The
	  update to 8.0 will fix /bin/csh references in /etc/passwd
	  and /etc/passwd.local to refer to /bin/athena/tcsh instead.

	* emacs18 has gone away.

	* AFS includes and libraries are installed in
	  /usr/afsws/include and /usr/afsws/lib.

	* wcl includes and libraries are installed in /usr/athena.

	* The basic krb5 headers and libraries are installed in
	  /usr/athena.

	* emacs has been upgraded from 19.30 to 19.34.  There should
	  be no significant user-visible changes.

	* RCS has been upgraded from version 4 to version 5.7.

	* x3270 has been upgraded from version 1.2 to version 3.1.1.3.

	* sysinfo is now available under Irix.

	* /bin/athena/tcsh now has Hesiod support enabled, which means
	  that "~lockername" will work in a pathname in most cases.
	  (It won't work for users with local passwd information who
	  don't have their lockers attached.)

	* For improved ease of debugging, many binaries are installed
	  with symbol tables, and compiled without optimization.
	  Therefore, the contents of /bin/athena and /usr/athena/bin
	  may be somewhat larger, and the binaries may consume more
	  CPU cycles, although it's unlikely that anyone will notice
	  the speed difference.


third/afs-krb5/src/aklog.c
/* 
 * $Id: aklog.c,v 1.1.1.1 2003-02-13 00:14:33 zacheiss Exp $
 *
 * Copyright 1990,1991 by the Massachusetts Institute of Technology
 * For distribution and copying rights, see the file "mit-copyright.h"
 */

#if !defined(lint) && !defined(SABER)
static char *rcsid = "$Id: aklog.c,v 1.1.1.1 2003-02-13 00:14:33 zacheiss Exp $";
#endif /* lint || SABER */

#include "aklog.h"

#ifndef WINDOWS

#ifdef __STDC__
main(int argc, char *argv[])
#else
main(argc, argv)
  int argc;
  char *argv[];
#endif /* __STDC__ */
{
    aklog_params params;

    aklog_init_params(&params);
    aklog(argc, argv, &params);
}

#else /* WINDOWS */

#include <windows.h>
#include <windowsx.h>

static void parse_cmdline();


int PASCAL
WinMain(HINSTANCE hinst, HINSTANCE hprevinstance, LPSTR cmdline, int noshow)
{
	int argc = 0;
	char **argv;

    aklog_params params;

	parse_cmdline(cmdline, &argv, &argc);

    aklog_init_params(&params);
    aklog(argc, argv, &params);

	return 0;
}

/*
 * Generate agrv/argc here from command line.
 * Note that windows doesn't pass us the executible name, so
 * we need to fill that in manually.
 */

static void
parse_cmdline(char *cmdline, char ***pargv, int *pargc)
{
	char **argv;
	int argc = 0;
	char *arg, *sep = " \t";
	int argv_size = 10;		/* to start with */


	argv = malloc(argv_size * sizeof(char *));

	if (!argv) {
		MessageBox(NULL, "Fatal Error: Out of memory", AKLOG_DIALOG_NAME, 
				   MB_OK | MB_ICONSTOP);
		exit(1);
	}

	argv[argc++] = "aklog";

	arg = strtok(cmdline, sep);

	while(arg) {
		argv[argc] = strdup(arg);

		if (!argv[argc]) {
			MessageBox(NULL, "Fatal Error: Out of memory", AKLOG_DIALOG_NAME, 
					   MB_OK | MB_ICONSTOP);
			exit(1);
		}

		argc++;

		if (argc == argv_size) {
			argv_size += 10;
			argv = realloc(argv, argv_size * sizeof(char *));

			if (!argv) {
				MessageBox(NULL, "Fatal Error: Out of memory",
						   AKLOG_DIALOG_NAME, 
						   MB_OK | MB_ICONSTOP);
				exit(1);
			}
		}

		arg = strtok(NULL, sep);
	}

	argv[argc] = NULL;

	*pargv = argv;
	*pargc = argc;
}

#endif /* WINDOWS */

